### PR TITLE
William/flow account

### PR DESCRIPTION
**Pull request by swansontec**
This is pretty chunky, but it improves typing in a lot of places. It's better to read the pull-request commit-by-commit.

### CODE REVIEW

**Comment by swansontec** [comment]
@wolverineks Can you take a look at this?

File: src/modules/login/login.js [hunk 1]
```diff
@@ -270,7 +281,11 @@ export function sanitizeLoginStash (stashTree: LoginStash, appId: string) {
     stash => stash.appId === appId,
     stash => stash,
     (stash, children) => {
-      const login = filterObject(stash, ['username', 'appId', 'loginId'])
+      const login: LoginStash = filterObject(stash, [
+        'username',
+        'appId',
+        'loginId'
+      ])
       login.children = children
```
Comment (paullinator) [inline src/modules/login/login.js]: Does this need an elvis statement like above?

File: src/modules/login/login.js [hunk 2]
```diff
@@ -253,8 +260,12 @@ export function makeLoginTree (
...
     stash => makeLoginTreeInner(stash, loginKey),
     (stash, children) => {
-      const login = filterObject(stash, ['username', 'appId', 'loginId'])
-      login.children = children
+      const login: LoginTree = filterObject(stash, [
+        'username',
+        'appId',
+        'loginId'
+      ])
+      login.children = children != null ? children : []
```
Comment (paullinator) [inline src/modules/login/login.js]: Was this a bug being fixed?

File: src/modules/account/accountApi.js [hunk 3]
```diff
@@ -87,7 +88,7 @@ function makeAccountApi (
 
     // Change or create credentials:
     changePassword (password: string): Promise<void> {
-      return state.changePassword(password)
+      return state.changePassword(password).then(() => {})
```
Comment (paullinator) [inline src/modules/account/accountApi.js]: What does this anonymous function do?

File: src/modules/account/account.test.js [hunk 4]
```diff
@@ -14,8 +14,8 @@ const contextOptions = {
   plugins: [makeFakeCurrency()]
 }
 
-function findKeys (keyInfos, type) {
-  return keyInfos.find(info => info.type === type)
+function findWallet (walletInfos, type) {
```
Comment (paullinator) [inline src/modules/account/account.test.js]: 😀👍

File: src/modules/login/login.js (hunk 2 above)
Comment (swansontec) [reply src/modules/login/login.js]: No, it's just typing. `children` can't be null in this context, but Flow doesn't know that, so I had to put in the check for the types to work. The fallback case should never be hit.

File: src/modules/login/login.js (hunk 1 above)
Comment (swansontec) [reply src/modules/login/login.js]: No. In this case, Flow is happy with things. `LoginStash.children` is allowed to be null.

File: src/modules/account/accountApi.js (hunk 3 above)
Comment (swansontec) [reply src/modules/account/accountApi.js]: It returns `undefined`, which is what Flow needs us to return.

**Comment by swansontec** [comment]
I'm tired of being blocked on this. Merging anyhow.

**Pull request closed without merging**